Configuration.md

```diff
@@ -91,29 +89,24 @@ These describe which data source to download, where the data should live, and ho
         * E.g. `['year', 'month']` will lead to a config set like `[(2015, 01), (2015, 02), (2015, 03), ...]`.
     * The list of keys will be used to format the `target_path`.
 
-### Creating a date-based directory hierarchy
-
-The configuration can be set up to automatically generate a date-based directory hierarchy for the output files.
+> **NOTE**: `target_path` and `target_filename` templates are totally compatible with Python's standard string formatting.
+> This includes being able to use named arguments (e.g. 'gs://bucket/{year}/{month}/{day}') as well as specifying formats for strings 
+> (e.g. 'gs://bucket/{year:04d}/{month:02d}/{day:02d}').
 
-To enable this feature, the `append_date_dirs` field has to be set to `true`. In addition, the `target_filename` needs
-to be specified, and `date` has to be a `partition_key`;
-`date` will not be used as a replacement in `target_template` but will instead be used to create a directory structure.
+### Creating a date-based directory hierarchy
 
-The resulting target path will be `<target_path>/{year}/{month}/{day}<target_filename>`. The number of format symbols in
-this path has to match the number of partition keys excluding `date`.
+The date-based directory hierarchy can be created using Python's standard string formatting.
+Below are some examples of how to use `target_path` and `target_filename` with Python's standard string formatting.
 
 <details>
 <summary><strong>Examples</strong></summary>
 
-Below are more examples of how to use `target_path`, `target_filename`, and `append_date_dirs`.
-
 Note that any parameters that are not relevant to the target path have been omitted.
 
 ```
 [parameters]
 target_filename=.nc
```

Comment:
I think we should get rid of `target_filename` too. What purpose does it serve that `target_path` does not?

Reply:
Also, related to #6, what is the best name for this field? `target_template`? Maybe I shouldn't overthink it and `target_path` is fine.

Reply:
`partition_template`?

Reply:
If we change the name, we should issue a deprecation warning to the user for now, and also accept it. 
Though, we shouldn't allow both names to be used at once.

Reply:
Yes, we can get rid of `target_filename`. I have removed the support for the same and pushed the code.

Regarding renaming of `target_path`, how about ` target_path_template` ? Also should we change it now ?